lib/web/imageRouter/index.js

```diff
@@ -15,12 +16,25 @@ const errors = require('../../errors')
 const imageRouter = (module.exports = Router())
 
 async function checkUploadType (filePath) {
-  const typeFromMagic = await FileType.fromFile(filePath)
+  const extension = path.extname(filePath).toLowerCase()
+  let typeFromMagic = await FileType.fromFile(filePath)
+  if (extension === '.svg') {
+    const fileContent = fs.readFileSync(filePath)
+    if (isSvg(fileContent)) {
+      typeFromMagic = {
+        ext: 'svg',
+        mime: 'image/svg+xml'
+      }
+    }
+  }
   if (typeFromMagic === undefined) {
     logger.error('Image upload error: Could not determine MIME-type')
     return false
   }
-  if (path.extname(filePath) !== '.' + typeFromMagic.ext) {
+  if (extension === '.jpeg' && typeFromMagic.ext === 'jpg') {
+    typeFromMagic.ext = 'jpeg'
+  }
+  if (extension !== '.' + typeFromMagic.ext) {
     logger.error(
       'Image upload error: Provided file extension does not match MIME-type'
     )
```

Comment:
Why do we check the extension of the uploaded file at all if we already check the magic number and could use the extension provided by file-info for storing the file instead?

Reply:
This fix was programmed in a way that does not rebuild the upload logic (and most likely introduce new bugs)…